src/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace preact {
 		type: ComponentType<P> | string;
 		props: P & { children: ComponentChildren };
 		key: Key;
-		ref: Ref<any> | null;
+		ref?: Ref<any> | null;
```

Comment:
Would you mind leaving a comment here on the importance of the `?` here? Since we don't have a test to verify we don't change this in the future, a comment stating why we have the `?` would be helpful

Reply:
@andrewiggins 
IMHO,
Changing type definition for removing error is replacing the means and purpose.
This change raises that error becomes normal.
For React compatibility, it is appropriate that `createVNode()` is changed, not type definition.
But, I think it is unnecessary.

Reply:
I updated the description and added a test project which demonstrates the error.

@38elements unnecessary? How are end users supposed to fix this issue? Even as a react lib author I would not know how to fix this issue?